ls1intum/Artemis
# added custom user login for admins

### Checklist

### Motivation and Context
Debugging statistics from students is hard because the rendered charts are calculated in the client for each student individually.

### Description
I added a button to login as another student for admins. When an admin logs in as another user the new user id is saved in the local storage and is used for the dashboard and result api calls. This gives the admin the option to debug the results of a user.

### Steps for Testing

1. Log in to ArTEMiS as Admin
2. Navigate to Student Dashboard
3. Select `Login as other Student`
4. Check if all results are for the selected user

### Screenshots
![image](https://user-images.githubusercontent.com/10998002/61086839-3b338e80-a434-11e9-9f4c-b038be2dcfb6.png)
![image](https://user-images.githubusercontent.com/10998002/61086848-44bcf680-a434-11e9-8916-d28743771825.png)
![image](https://user-images.githubusercontent.com/10998002/61086857-4ab2d780-a434-11e9-81e9-14630ef89994.png)
